includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -682,6 +682,24 @@ public function send_failed_order_email( $order_id, $status_update = [] ) {
 		$emails['WC_Email_Failed_Order']->trigger( $order_id );
 	}
 
+	/**
+	 * Sends the failed refund email to both admin and shopper.
+	 *
+	 * @since 9.6.0
+	 * @param WC_Order $order The order object for which the refund failed.
+	 * @return void
+	 */
+	public function send_failed_refund_emails( $order ) {
+		$emails = WC()->mailer()->get_emails();
+
+		if ( empty( $emails ) || empty( $order ) || ! isset( $emails['WC_Stripe_Email_Admin_Failed_Refund'] ) || ! isset( $emails['WC_Stripe_Email_Customer_Failed_Refund'] ) ) {
+			return;
+		}
+
+		$emails['WC_Stripe_Email_Admin_Failed_Refund']->trigger( $order->get_id(), $order );
+		$emails['WC_Stripe_Email_Customer_Failed_Refund']->trigger( $order->get_id(), $order );
```

Comment:
We should split the conditions up a bit here -- as things are written, we won't send any emails if one of the values is not set.
```suggestion
		if ( empty( $emails ) || empty( $order ) ) {
			return;
		}

		if ( isset( $emails['WC_Stripe_Email_Admin_Failed_Refund'] ) ) {
			$emails['WC_Stripe_Email_Admin_Failed_Refund']->trigger( $order->get_id(), $order );
		}

		if ( isset( $emails['WC_Stripe_Email_Customer_Failed_Refund'] ) ) {
			$emails['WC_Stripe_Email_Customer_Failed_Refund']->trigger( $order->get_id(), $order );
		}
```